### PR TITLE
refactor: override HasStyle implementation in LoginOverlay

### PR DESCRIPTION
**Pull request by web-padawan**
## Description

Same as #4549 and #4576 but for `LoginOverlay`.

Based on #4573

## Type of change

- Refactor

### CODE REVIEW

File: vaadin-login-flow-parent/vaadin-login-flow/src/test/java/com/vaadin/flow/component/login/LoginOverlayHasStyleTest.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
+package com.vaadin.flow.component.login;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.component.html.Div;
```
Comment (sissbruecker) [inline vaadin-login-flow-parent/vaadin-login-flow/src/test/java/com/vaadin/flow/component/login/LoginOverlayHasStyleTest.java:10]:
```suggestion
import com.vaadin.flow.component.UI;
```
Unused import.

File: vaadin-login-flow-parent/vaadin-login-flow/src/main/java/com/vaadin/flow/component/login/LoginOverlay.java [hunk 2]
```diff
@@ -231,6 +231,27 @@ public String getDescription() {
         return getElement().getProperty("description");
     }
 
+    /**
+     * Sets the CSS class names of the dialog overlay element. This method
```
Comment (sissbruecker) [inline vaadin-login-flow-parent/vaadin-login-flow/src/main/java/com/vaadin/flow/component/login/LoginOverlay.java:235]:
```suggestion
     * Sets the CSS class names of the login overlay element. This method
```

**Review by sissbruecker — APPROVED** [review]

File: vaadin-login-flow-parent/vaadin-login-flow/pom.xml [hunk 3]
```diff
@@ -39,6 +39,11 @@
             <artifactId>flow-test-generic</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>com.vaadin</groupId>
+            <artifactId>flow-html-components</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (sissbruecker) [inline vaadin-login-flow-parent/vaadin-login-flow/pom.xml:46]: Seems this is not needed anymore after removing the unused import.

Comment (web-padawan) [reply vaadin-login-flow-parent/vaadin-login-flow/pom.xml]: Thanks, removed.

**Pull request merged**